src/SalmonDE/StatsPE/FloatingTexts/EventListener.php

```diff
@@ -5,14 +5,21 @@
 use pocketmine\event\Listener;
 use pocketmine\event\player\PlayerJoinEvent;
 use pocketmine\Player;
+use SalmonDE\StatsPE\StatsBase;
 
 class EventListener implements Listener {
 
+    private $owner;
+
+    public function __construct(StatsBase $owner){
+        $this->owner = $owner;
+    }
+
     /**
      * @priority MONITOR
      */
     public function onJoin(PlayerJoinEvent $event){
-        $floatingTexts = FloatingTextManager::getInstance()->getAllFloatingTexts();
+        $floatingTexts = $this->owner->getFloatingTextManager()->getAllFloatingTexts();
         if(isset($floatingTexts[$event->getPlayer()->getLevel()->getFolderName()])){
             foreach($floatingTexts[$event->getPlayer()->getLevel()->getFolderName()] as $floatingText){
                 $floatingText->sendTextToPlayer($event->getPlayer());
```

Comment:
I'm not sure how to reference `FloatingText.php`, where `sendTextToPlayer()` is located because you need to keep `$floatingText` to send to those people.